Code/Framework/AzFramework/Tests/QualitySystemComponentTests.cpp

```diff
@@ -0,0 +1,242 @@
+            AZ::UserSettingsComponentRequestBus::Broadcast(&AZ::UserSettingsComponentRequests::DisableSaveOnFinalize);
+        }
+
+        AZ::IConsole* m_console;
+        AZStd::unique_ptr<AzFramework::Application> m_application;
+        AZStd::unique_ptr<AZ::SettingsRegistryInterface> m_settingsRegistry;
+    };
+
+    TEST_F(QualitySystemComponentTestFixture, QualitySystem_Registers_Group_CVars)
+    {
+        // when the quality system component registers group cvars
+        StartApplicationWithSettings(R"(
+            {
+                "O3DE": {
+                    "Quality": {
+                        "Groups": {
+                            "q_test": {
+                                "Levels": [ "low", "high" ],
+                                "Default": 1,
+                                "Description": "q_test quality group",
+                                "Settings": {
+                                    "q_test_sub": [0,1]
+                                }
+                            },
+                            "q_test_sub": {
+                                "Levels": [ "low", "high" ],
+                                "Default": 0,
+                                "Description": "q_test_sub quality group",
+                                "Settings": {
+                                    "a_cvar": [123,234]
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            )");
+
+        // expect the cvars are created with their default values
+        auto value = AzFramework::QualityLevel::LevelFromDeviceRules;
+        EXPECT_EQ(m_console->GetCvarValue("q_test", value), AZ::GetValueResult::Success);
+        EXPECT_EQ(value, AzFramework::QualityLevel{1});
+
+        EXPECT_EQ(m_console->GetCvarValue("q_test_sub", value), AZ::GetValueResult::Success);
+        EXPECT_EQ(value, AzFramework::QualityLevel::DefaultQualityLevel);
+    }
+
+    AZ_CVAR(int32_t, a_setting, 0, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Example integer setting 1");
+    AZ_CVAR(AZ::CVarFixedString, b_setting, "default", nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Example string setting 2");
+    AZ_CVAR(int32_t, c_setting, -1, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Example integer setting 3");
+    AZ_CVAR(int32_t, d_setting, -2, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Example integer setting 4");
+
+    TEST_F(QualitySystemComponentTestFixture, QualitySystem_Loads_Group_Level)
+    {
+        // when the quality system component registers group cvars
+        StartApplicationWithSettings(R"(
+            {
+                "O3DE": {
+                    "Quality": {
+                        "DefaultGroup":"q_test",
+                        "Groups": {
+                            "q_test": {
+                                "Levels": [ "low", "medium","high", "veryhigh"],
```

Comment:
extreme nitpick: inconsistent spacing between medium and high
```suggestion
                                "Levels": [ "low", "medium", "high", "veryhigh"],
```